src/modules/user-auth/utils/AuthGuard/AuthGuard.tsx

```diff
@@ -0,0 +1,38 @@
+import { FC, ReactNode, useEffect } from 'react';
+import { useAuth } from '@modules/user-auth/contexts/AuthContext';
+import DefaultUnauthenticatedPrompt from '../../components/DefaultUnauthenticatedPrompt';
+import DefaultEmailUnverifiedPrompt from '../../components/DefaultEmailUnverifiedPrompt';
+import { useLayoutModal } from '@modules/layouts/contexts/LayoutModalContext';
+
+type AuthGuardProps = {
+  emailVerified?: boolean;
+  promptLogin?: boolean;
+  renderUnauthenticated?: ReactNode;
+  renderEmailUnverified?: ReactNode;
+};
+
+const AuthGuard: FC<AuthGuardProps> = ({
+  children,
+  emailVerified = false,
+  promptLogin = false,
```

Comment:
So there are pages where they are wrapped inside AuthGuard but don't require prompting login?

Reply:
I imagine we can reuse the `AuthGuard` not just for the entire page but for area in a page if we want to show an alternative content in that area depending on the auth state. We can make it true by default so we don't have to think about it.

Reply:
We should make a different component for that case. I don't see any similar logic that will be shared between page level and component level.

Reply:
Here's an example use case:

```jsx
<AuthGuard
    renderedUnauthenticatred={<button disabled>Login to  contact owner</button>}
>
<button>Contact property owner</button>
</AuthGuard>
```

We are currently using this in the layouts which is the componenet level anyway, just that the layout ys wrapping around a page

Reply:
Also,I think we only think of guard at the page level before is because the concept comes from Angular